### DISCUSSION TITLE
Somatic hypermutation analysis

### DISCUSSION THREAD

**Discussion by pbpayal** [milaboratory/mixcr#1660] [Q&A] [answered]
How to do somatic hypermutation analysis between two sample sets (3 samples per cohort) Can we do somatic hypermutation analysis from https SMARTer Mouse TCR a/b Profiling Kit? I came across "B cell lineage tracing" page. 

How to create a diagram for mutation frequencies between two cohorts( no timepoints)?

**Comment by mizraelson** [comment]
Do you expect somatic hypermutations to be present in TCRs?

**Reply by pbpayal** [reply]
Extremely sorry. My mistake..I put the wrong kit. This is BCR sequencing data. (SMARTer Mouse BCR IgG H/K/L Profiling Kit)

**Reply by pbpayal** [reply]
I guess this is a very broad question. I am not an expert in immunology, so pardon my lack of in depth knowledge in this section. I see in the tutorial, a specific tree id is being chosen to do the tree reconstruction(https://mixcr.com/mixcr/guides/ig-trees-reconstruction/?h=trees#export-linage-trees-in-a-graphical-format). Whats the criteria to choose that specific tree id?

I have extracted the trees. What am I supposed to do after that? I want to look at the differences in WT vs Knockout mice?
`
mixcr findAlleles --report findAlleles.report.txt --export-alleles-mutations alleles.tsv --export-library alleles.json --output-template {file_dir_path}/{file_name}.group.reassigned.clns *.clns

mixcr findShmTrees --report Part1_trees.log *reassigned.clns Part1.shmt

mixcr exportShmTreesWithNodes part1_trees.shmt  part1_trees.txt

`

**Reply by pbpayal** [reply]
And when I tried doing a random tree node, I get this error:

`mixcr exportPlots shmTrees -m metadata_part1.tsv --node-color condition  --ids 1424 part1_trees.shmt Mutant_shm_trees_1424.pdf`

**Error while running command shmTrees java.lang.RuntimeException: Duplicate library: repseqio.v4.0_with_found_alleles:10090**
 
Attached metadata file.
[metadata_part1.txt](https://github.com/milaboratory/mixcr/files/15312762/metadata_part1.txt)

**Reply by mizraelson** [reply, marked as answer]
Hi, sorry for the late reply.

To compare the SHMs status between two groups I suggest first run `findAlleles` and then:

 ```
 mixcr exportClones \
-nMutationRate VRegionTrimmed \
-nMutationRate JRegionTrimmed \
input.clns output.tsv
```

This will return a repertoire table with the SHM rate (number of mutations divided by the gene segment length) for V and J genes. Then, you can take the average number across all clones in each sample and compare the two groups.

The trees make sense only within the same organism. I wouldn't recommend plotting a tree for multiple mice. The trees are rather used to track the B cell lineage over time (e.g., after vaccination or another event) in the same organism. In that case, the general approach is to pick a tree with a large number of nodes and look for one that has long branches.

**Reply by pbpayal** [reply]
Thank you for detailed explanation.
